ormar/queryset/join.py

```diff
@@ -92,28 +92,29 @@ def to_table(self) -> sqlalchemy.Table:
         """
         return self.next_model.Meta.table
 
-    def _on_clause(self, previous_alias: str, from_clause: str, to_clause: str) -> text:
+    def _on_clause(self, previous_alias: str, from_table_name:str, from_column_name: str, to_table_name: str, to_column_name: str) -> text:
         """
         Receives aliases and names of both ends of the join and combines them
         into one text clause used in joins.
 
         :param previous_alias: alias of previous table
         :type previous_alias: str
-        :param from_clause: from table name
-        :type from_clause: str
+        :param from_table_name: from table name
+        :type from_table_name: str
+        :param from_column_name: from column name
+        :type from_column_name: str
         :param to_clause: to table name
         :type to_clause: str
         :return: clause combining all strings
         :rtype: sqlalchemy.text
         """
-        left_part = f"{self.next_alias}_{to_clause}"
+        dialect = self.main_model.Meta.database._backend._dialect
+        quoter = dialect.identifier_preparer.quote
+        left_part = f"{quoter(self.next_alias + '_' + to_table_name)}.{quoter(to_column_name)}"
```

Comment:
Please replace string concat with nested f-string
`left_part = f"{quoter(f'{self.next_alias}_{to_table_name}')}.{quoter(to_column_name)}"`

Reply:
Ah I thought that was going to be a 3.12 thing lol